From 15230447a933117cd4da896cb896255b796abba1 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Mon, 8 Jun 2026 16:21:31 +0200
Subject: [PATCH] scripts/ext-toolchain.sh: Start with versioned binaries

Make sure that when searching for CC, we will start with versioned
binaries and fallback to other binaries only later on. Otherwise we
might end up with arm-openwrt-linux-muslgnueabi-gcc-ar being our CC.
---
 scripts/ext-toolchain.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/ext-toolchain.sh b/scripts/ext-toolchain.sh
index e49c011118..720c452ec7 100755
--- a/scripts/ext-toolchain.sh
+++ b/scripts/ext-toolchain.sh
@@ -453,7 +453,7 @@ probe_cc() {
 		local bin
 		for bin in "bin" "usr/bin" "usr/local/bin"; do
 			local cmd
-			for cmd in "$TOOLCHAIN/$bin/"*-*cc*; do
+			for cmd in "$TOOLCHAIN/$bin/"*-*cc*[0-9] "$TOOLCHAIN/$bin/"*-*cc*; do
 				if [ -x "$cmd" ] && [ ! -h "$cmd" ]; then
 					CC="$(cd "${cmd%/*}"; pwd)/${cmd##*/}"
 					return 0
-- 
2.54.0

